python/mdvtools/socketio_upload.py

```
@@ -0,0 +1,697 @@
+    # Get the SocketIO instance from the app extensions
+    # This assumes mdv_socketio(app) was called first
+    if hasattr(app, 'extensions') and 'socketio' in app.extensions:
+        _socketio_instance = app.extensions['socketio']
+    else:
+        # Fallback: try to import from mdvtools.websocket
+        try:
+            from mdvtools.websocket import socketio as imported_socketio
+            _socketio_instance = imported_socketio
+            upload_log("Retrieved SocketIO instance from mdvtools.websocket")
+        except ImportError:
+            raise RuntimeError("SocketIO not found. Make sure mdv_socketio(app) is called before initialize_socketio_upload(app)")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)